### ISSUE TITLE
dark world with shaders and lots of other problems when using shaders

### ISSUE THREAD

**Issue by silverace71** [AnOpenSauceDev/Methane-mod#12]
# using any shaders with the mod makes the world dark

Just to be clear, great mod, great performance increse (I went from 700 fps up to 2200 fps) but using shaders for some reason just breaks everything visually. So no crashes but lots of problems. I will have lots of pictures and logs below for you to look at! Also you can contact me on discord at silverace_71#3430 If you need more testing/images/logs. I really like this mod due to its performance

## What mods am I using?
```
[15:52:56] [main/INFO]: Loading 267 mods:
	- accurateblockplacement 1.0.14
	- advanced_runtime_resource_pack 0.6.2 via enhancedblockentities
	- advancedchatcore 1.19.2-1.5.8
	- advancedchatlog 1.19-1.2.2
	- ambientsounds 5.2.5
	- amecs 1.3.8+mc.1.19-rc2
	- amecsapi 1.3.6+mc22w17a via amecs
	- animatica 0.5+1.19
	- appleskin 2.4.1+mc1.19
	- architectury 6.3.49
	- auditory 0.0.3-1.19.x
	- auudio 1.0.3
	- bactromod 1.0
	- betteranimationscollection 4.0.5
	- betterbiomeblend 1.19-1.3.6-fabric
	- betterf3 1.4.0
	- bettermounthud 1.2.0
	- betterpingdisplay 1.1.1
	- betterstats 1.4.3
	- blanket-client-tweaks 1.1.2
	- borderlessmining 1.1.5+1.19
	- buildguide 0.3.3
	- c2me 0.2.0+alpha.9.0
	- c2me-base 0.2.0+alpha.9.0 via c2me
	- c2me-client-uncapvd 0.2.0+alpha.9.0 via c2me
	- c2me-fixes-chunkio-threading-issues 0.2.0+alpha.9.0 via c2me
	- c2me-fixes-general-threading-issues 0.2.0+alpha.9.0 via c2me
	- c2me-fixes-worldgen-threading-issues 0.2.0+alpha.9.0 via c2me
	- c2me-fixes-worldgen-vanilla-bugs 0.2.0+alpha.9.0 via c2me
	- c2me-notickvd 0.2.0+alpha.9.0 via c2me
	- c2me-opts-allocs 0.2.0+alpha.9.0 via c2me
	- c2me-opts-chunk-access 0.2.0+alpha.9.0 via c2me
	- c2me-opts-chunkio 0.2.0+alpha.9.0 via c2me
	- c2me-opts-math 0.2.0+alpha.9.0 via c2me
	- c2me-opts-scheduling 0.2.0+alpha.9.0 via c2me
	- c2me-opts-worldgen-general 0.2.0+alpha.9.0 via c2me
	- c2me-opts-worldgen-vanilla 0.2.0+alpha.9.0 via c2me
	- c2me-rewrites-chunkio 0.2.0+alpha.9.0 via c2me
	- c2me-server-utils 0.2.0+alpha.9.0 via c2me
	- c2me-threading-chunkio 0.2.0+alpha.9.0 via c2me
	- c2me-threading-lighting 0.2.0+alpha.9.0 via c2me
	- c2me-threading-scheduling 0.2.0+alpha.9.0 via c2me
	- c2me-threading-worldgen 0.2.0+alpha.9.0 via c2me
	- c3h6n6o6 1.0.0-beta9-1.18.2
	- caffeineconfig 1.0.0+1.17 via sodium-extra
	- camera_keybinds 2.0
	- cameraoverhaul 1.3.1-fabric-universal
	- capes 1.4.5+1.19
	- cardinal-components-base 5.0.2 via puzzleslib
	- cardinal-components-block 5.0.2 via puzzleslib
	- cardinal-components-chunk 5.0.2 via puzzleslib
	- cardinal-components-entity 5.0.2 via puzzleslib
	- cardinal-components-world 5.0.2 via puzzleslib
	- carpet 1.4.84
	- carpet-fixes 1.19-1.12.0
	- cem 0.7.1
	- chime 1.3.0
	- citresewn 1.1.2+1.19.2
	- citresewn-defaults 1.1.2+1.19.2 via citresewn
	- cloth-basic-math 0.6.1 via cloth-config
	- cloth-config 8.2.88
	- coat 1.0.0-beta.19+mc22w17a via completeconfig
	- colormatic 3.1.2+mc.1.19
	- com_electronwill_night-config_core 3.6.5 via advancedchatcore
	- com_electronwill_night-config_json 3.6.5 via darkkore
	- com_electronwill_night-config_toml 3.6.5 via advancedchatcore
	- com_eliotlash_mclib_mclib 20 via geckolib3
	- com_github_darkkronicle_konstruct_addons 2.0.3-build1 via advancedchatcore
	- com_github_darkkronicle_konstruct_core 2.0.3-build1 via advancedchatcore
	- com_github_llamalad7_mixinextras 0.1.1 via fastquit
	- com_github_luben_zstd-jni 1.5.2-2 via dashloader
	- com_googlecode_soundlibs_mp3spi 1.9.5-1 via craftgr
	- com_googlecode_soundlibs_tritonus-share 0.3.7-2 via craftgr
	- com_ibm_async_asyncutil 0.1.0 via c2me
	- com_logisticscraft_occlusionculling 0.0.6-SNAPSHOT via entityculling
	- com_velocitypowered_velocity-native 3.1.2-SNAPSHOT via krypton
	- completeconfig 2.1.0
	- completeconfig-base 2.1.0 via completeconfig
	- completeconfig-gui-cloth 2.1.0 via completeconfig
	- completeconfig-gui-coat 2.1.0 via completeconfig
	- completeconfig-gui-yacl 2.1.0 via completeconfig
	- conditional-mixin 0.3.0 via carpet-fixes
	- continuity 2.0.1+1.19
	- craftgr 1.4.1-mc1.19.2-fabric
	- creativecore 2.8.5
	- cull-less-leaves 1.0.6
	- cullclouds 0.1.0
	- cullleaves 2.3.4
	- darkkore 0.3.1-1.19
	- debugify 2.7.1
	- dev_quantumfusion_hyphen 0.4.0-rc.3 via dashloader
	- dev_quantumfusion_taski 2.1.0 via dashloader
	- do_a_barrel_roll 2.2.0+1.19.2
	- double_hotbar 1.2.2
	- drippyloadingscreen 1.6.5
	- dynamiccrosshair 3.9+1.19
	- dynamiccrosshaircompat 1.9+1.19.2
	- enhancedblockentities 0.7.2+1.19.2
	- entity_texture_features 4.1.1
	- entitycollisionfpsfix 2.0.0.0
	- entityculling 1.5.2-mc1.19
	- error_notifier 1.0.9 via roughlyenoughitems
	- exordium 1.0.2-mc1.19.2
	- extrasounds 2.3.1+1.19.2-1.19.1
	- fabric-api 0.67.1+1.19.2
	- fabric-api-base 0.4.13+0d0f210290 via fabric-api
	- fabric-api-lookup-api-v1 1.6.12+93d8cb8290 via fabric-api
	- fabric-biome-api-v1 9.0.20+0d0f210290 via fabric-api
	- fabric-block-api-v1 1.0.1+12bfe4ea90 via fabric-api
	- fabric-blockrenderlayer-v1 1.1.22+c6af733c90 via fabric-api
	- fabric-client-tags-api-v1 1.0.3+b35fea8390 via fabric-api
	- fabric-command-api-v1 1.2.13+f71b366f90 via fabric-api
	- fabric-command-api-v2 2.1.9+93d8cb8290 via fabric-api
	- fabric-commands-v0 0.2.30+df3654b390 via fabric-api
	- fabric-containers-v0 0.1.37+df3654b390 via fabric-api
	- fabric-content-registries-v0 3.4.3+6064db2a90 via fabric-api
	- fabric-convention-tags-v1 1.1.3+93d8cb8290 via fabric-api
	- fabric-crash-report-info-v1 0.2.7+aeb40ebe90 via fabric-api
	- fabric-data-generation-api-v1 5.3.1+ad31814d90 via fabric-api
	- fabric-dimensions-v1 2.1.33+0d0f210290 via fabric-api
	- fabric-entity-events-v1 1.5.2+9244241690 via fabric-api
	- fabric-events-interaction-v0 0.4.30+c6af733c90 via fabric-api
	- fabric-events-lifecycle-v0 0.2.31+df3654b390 via fabric-api
	- fabric-game-rule-api-v1 1.0.23+c6af733c90 via fabric-api
	- fabric-item-api-v1 1.6.1+fa140d5990 via fabric-api
	- fabric-item-groups-v0 0.3.35+9244241690 via fabric-api
	- fabric-key-binding-api-v1 1.0.23+aaaf9d3390 via fabric-api
	- fabric-keybindings-v0 0.2.21+df3654b390 via fabric-api
	- fabric-language-kotlin 1.8.5+kotlin.1.7.20
	- fabric-lifecycle-events-v1 2.2.2+9244241690 via fabric-api
	- fabric-loot-api-v2 1.1.9+83a8659290 via fabric-api
	- fabric-loot-tables-v1 1.1.12+9e7660c690 via fabric-api
	- fabric-message-api-v1 5.0.5+93d8cb8290 via fabric-api
	- fabric-mining-level-api-v1 2.1.20+33fbc73890 via fabric-api
	- fabric-models-v0 0.3.19+c6af733c90 via fabric-api
	- fabric-networking-api-v1 1.2.7+9244241690 via fabric-api
	- fabric-networking-v0 0.3.24+df3654b390 via fabric-api
	- fabric-object-builder-api-v1 4.1.2+06e0e25190 via fabric-api
	- fabric-particles-v1 1.0.12+79adfe0a90 via fabric-api
	- fabric-registry-sync-v0 0.9.28+9244241690 via fabric-api
	- fabric-renderer-api-v1 1.0.14+12bfe4ea90 via fabric-api
	- fabric-renderer-indigo 0.6.16+aeb40ebe90 via fabric-api
	- fabric-renderer-registries-v1 3.2.22+df3654b390 via fabric-api
	- fabric-rendering-data-attachment-v1 0.3.17+0d0f210290 via fabric-api
	- fabric-rendering-fluids-v1 3.0.9+c6af733c90 via fabric-api
	- fabric-rendering-v0 1.1.25+df3654b390 via fabric-api
	- fabric-rendering-v1 1.11.1+73145abb90 via fabric-api
	- fabric-resource-conditions-api-v1 2.1.1+aae9039d90 via fabric-api
	- fabric-resource-loader-v0 0.8.3+251d254a90 via fastload
	- fabric-screen-api-v1 1.0.29+0d0f210290 via fabric-api
	- fabric-screen-handler-api-v1 1.3.3+1cc24b1b90 via fabric-api
	- fabric-sound-api-v1 1.0.1+c4f28df590 via fabric-api
	- fabric-textures-v0 1.0.22+aeb40ebe90 via fabric-api
	- fabric-transfer-api-v1 2.1.3+9d081d6490 via fabric-api
	- fabric-transitive-access-wideners-v1 1.3.2+42d99c3290 via fabric-api
	- fabrication 3.0.3+1.19
	- fabricloader 0.14.10
	- fancymenu 2.12.9
	- faster_entity_animations 1.5.7
	- fastload 2.6.10
	- fastquit 2.1.0+1.19-1.19.2
	- ferritecore 4.2.1
	- feytweaks 1.19-1.2.3
	- forgeconfigapiport 4.2.7
	- forgetmechunk 1.0.4-1.18.X-1.19.X
	- fvt 1.13.0-mc1.19.2
	- geckolib3 3.1.37
	- guardian 1.0-SNAPSHOT
	- hud-moving 1.0
	- hudium 1.4.5+mc1.19
	- ias 8.0.1
	- immediatelyfast 1.1.6
	- improved-signs 1.2.0
	- indium 1.0.9+mc1.19.2
	- inventoryhud 3.4.2
	- io_github_douira_glsl-transformer 2.0.0-pre8 via iris
	- io_github_maowimpl_owo 2.0.0 via advancedchatcore
	- iris 1.5.0
	- itemmodelfix 1.0.3+1.19
	- java 17
	- javazoom_jlayer 1.0.1 via craftgr
	- kirin 1.11.1 via presencefootsteps
	- konkrete 1.5.3
	- koroworld-lib 2.0.1
	- kronhud 1.19.2-2.2.3
	- krypton 0.2.1
	- lambdabettergrass 1.3.0+1.19
	- lambdynlights 2.1.2+1.19
	- lazydfu 0.1.3
	- lib39-deferral 1.1.1 via fabrication
	- lithium 0.10.0
	- loadingscreentips 1.1.0
	- loadmyresources 1.0.3
	- lod 1.6.9a
	- logs-be-gone 1.1.0
	- malilib 0.13.0
	- mbp 1.4
	- memoryleakfix 1.19.1-0.7.0
	- methane 1.4
	- midnightlib 0.6.0 via loadingscreentips
	- minecraft 1.19.2
	- minihud 0.23.3
	- modmenu 4.0.6
	- moreculling 1.19.1-0.14.0
	- moremcmeta 1.19-3.0.5
	- moremusic 0.1.3+1.19
	- mousewheelie 1.10.2+mc1.19-pre1
	- net_objecthunter_exp4j 0.4.8 via c2me
	- notenoughanimations 1.6.1
	- noxesium 0.1.2
	- omega-config 1.3-1.19-pre5 via hudium
	- optigui 1.1.6
	- org_anarres_jcpp 1.4.14 via iris
	- org_antlr_antlr4-runtime 4.11.1 via iris
	- org_apache_commons_commons-csv 1.8 via advancedchatcore
	- org_apache_commons_commons-text 1.9 via optigui
	- org_apache_httpcomponents_httpmime 4.5.10 via entity_texture_features
	- org_jetbrains_kotlin_kotlin-reflect 1.7.20 via fabric-language-kotlin
	- org_jetbrains_kotlin_kotlin-stdlib 1.7.20 via fabric-language-kotlin
	- org_jetbrains_kotlin_kotlin-stdlib-jdk7 1.7.20 via fabric-language-kotlin
	- org_jetbrains_kotlin_kotlin-stdlib-jdk8 1.7.20 via fabric-language-kotlin
	- org_jetbrains_kotlinx_atomicfu-jvm 0.18.4 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-coroutines-core-jvm 1.6.4 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-coroutines-jdk8 1.6.4 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-datetime-jvm 0.4.0 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-serialization-cbor-jvm 1.4.1 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-serialization-core-jvm 1.4.1 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-serialization-json-jvm 1.4.1 via fabric-language-kotlin
	- org_joml_joml 1.10.4 via sodium
	- org_mariuszgromada_math_mathparser_org-mxparser 4.4.2 via advancedchatcore
	- org_threadly_threadly 7.0 via c2me
	- presencefootsteps 1.6.2
	- pride 1.1.2+1.19 via fabrication
	- puzzle 1.4.0-1.19
	- puzzle-base 1.4.0-1.19 via puzzle
	- puzzle-gui 1.4.0-1.19 via puzzle
	- puzzle-models 1.4.0-1.19 via puzzle
	- puzzle-splashscreen 1.4.0-1.19 via puzzle
	- puzzleslib 4.3.12
	- reeses-sodium-options 1.4.7+mc1.19.2-build.59
	- roughlyenoughitems 9.1.555
	- shulkerboxtooltip 3.1.3+1.19.2
	- simplefog 1.3.1
	- skinlayers 1.5.2-mc1.19.1
	- smoothboot 1.19-1.7.1
	- sneaktweak 1.8.0
	- sodium 0.4.4+build.18
	- sodium-extra 0.4.13+mc1.19.2-build.76
	- sound_physics_remastered 1.19.2-1.0.15
	- soundcategories 1.2.4+1.19 via extrasounds
	- spruceui 4.0.0+1.19 via enhancedblockentities
	- sspb 3.1.0
	- tooltipfix 1.1.1-1.19
	- transparent 5.1.2
	- tweed4_annotated 1.3.1+mc22w17a via mousewheelie
	- tweed4_base 1.6.0+mc22w17a via mousewheelie
	- tweed4_data 1.2.1+mc22w17a via mousewheelie
	- tweed4_data_hjson 1.1.1+mc22w17a via mousewheelie
	- tweed4_tailor_coat 1.1.3+mc22w17a via mousewheelie
	- tweed4_tailor_screen 1.1.1+mc22w17a via mousewheelie
	- vmp 0.2.0+beta.7.11
	- wi_zoom 1.4-MC1.19.2
	- worldedit 7.2.12+6240-87f4ae1
	- xaerominimap 22.16.0
	- xaeroworldmap 1.28.1
	- yet-another-config-lib 1.7.1
```
## Things that work with shaders
- no shaders
- emmisive lighting (kinda)
- anywhere that has 15 light level looks normal (any light emmiting block)
## Things that dont work
- shadows
- emmisive lighting in shadow borders is weird. It both lights up and darkens the surrounding area
- using any shaders I get the same error
```
[15:39:57] [Render thread/WARN]: Failed to resolve uniform rainStrength, reason: Variable shadows build in uniform: rainStrength ( = FunctionCall{clamp {[Id{rainStrength}, Number{0.0}, Number{1.0}]} })
java.lang.Exception: Variable shadows build in uniform: rainStrength
	at net.coderbot.iris.uniforms.custom.CustomUniforms.addVariable(CustomUniforms.java:175) ~[iris-mc1.19.2-1.5.0.jar:?]
	at net.coderbot.iris.uniforms.custom.CustomUniforms.<init>(CustomUniforms.java:56) ~[iris-mc1.19.2-1.5.0.jar:?]
	at net.coderbot.iris.uniforms.custom.CustomUniforms$Builder.build(CustomUniforms.java:317) ~[iris-mc1.19.2-1.5.0.jar:?]
	at net.coderbot.iris.uniforms.custom.CustomUniforms$Builder.build(CustomUniforms.java:329) ~[iris-mc1.19.2-1.5.0.jar:?]
	at net.coderbot.iris.pipeline.newshader.NewWorldRenderingPipeline.<init>(NewWorldRenderingPipeline.java:215) ~[iris-mc1.19.2-1.5.0.jar:?]
	at net.coderbot.iris.Iris.createPipeline(Iris.java:656) ~[iris-mc1.19.2-1.5.0.jar:?]
	at net.coderbot.iris.pipeline.PipelineManager.preparePipeline(PipelineManager.java:34) [iris-mc1.19.2-1.5.0.jar:?]
	at net.coderbot.iris.Iris.reload(Iris.java:598) [iris-mc1.19.2-1.5.0.jar:?]
	at net.coderbot.iris.Iris.toggleShaders(Iris.java:265) [iris-mc1.19.2-1.5.0.jar:?]
	at net.coderbot.iris.Iris.handleKeybinds(Iris.java:246) [iris-mc1.19.2-1.5.0.jar:?]
	at net.minecraft.class_310.handler$fol000$iris$onTick(class_310.java:21460) [client-intermediary.jar:?]
	at net.minecraft.class_310.method_1574(class_310.java:1907) [client-intermediary.jar:?]
	at net.minecraft.class_310.method_1523(class_310.java:1135) [client-intermediary.jar:?]
	at net.minecraft.class_310.method_1514(class_310.java:768) [client-intermediary.jar:?]
	at net.minecraft.client.main.Main.method_44604(Main.java:244) [client-intermediary.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:51) [client-intermediary.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461) [fabric-loader-0.14.10.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.14.10.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) [fabric-loader-0.14.10.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:210) [NewLaunch.jar:?]
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:245) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.listen(EntryPoint.java:143) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.main(EntryPoint.java:34) [NewLaunch.jar:?]
[15:39:57] [Render thread/INFO]: Stopping worker threads
[15:39:57] [Render thread/INFO]: Started 12 worker threads
[15:39:57] [Render thread/INFO]: [CHAT] Toggled shaders to ComplementaryShaders_v4.6.zip!
[15:39:57] [Render thread/INFO]: [15:39:57] Toggled shaders to ComplementaryShaders_v4.6.zip!
[15:39:57] [Render thread/INFO]: [System] [CHAT] Toggled shaders to ComplementaryShaders_v4.6.zip!
```
- Also this popped up randomly
```
[16:08:49] [Render thread/ERROR]: ########## GL ERROR ##########
[16:08:49] [Render thread/ERROR]: @ Render
[16:08:49] [Render thread/ERROR]: 65540: Invalid scancode -1
```
## Pictures!
- This is normal
![2023-01-14_16 07 34](https://user-images.githubusercontent.com/77552030/212500974-b2f0bacb-5697-4e03-b5c5-4e63c2dffc79.png)
- This is with shaders, without moving around
![2023-01-14_16 09 14](https://user-images.githubusercontent.com/77552030/212501028-790c4877-5064-4f4b-be0b-9a0650e60727.png)
- This is after rotating 360 and redering everything
![2023-01-14_16 10 24](https://user-images.githubusercontent.com/77552030/212501045-abe4d5ac-b49d-45a8-ad30-9d7d1cd94a6d.png)
- emmisive lighting
![2023-01-14_16 11 36](https://user-images.githubusercontent.com/77552030/212501092-48bed6ae-4a72-4d88-bf16-2909a95b18f1.png)
- doing stuff with light blocks
![2023-01-14_16 13 47](https://user-images.githubusercontent.com/77552030/212501126-8cd66a00-fe7b-46e6-afa8-c6568c4c0372.png)

**Comment by AnOpenSauceDev** [comment]
Some shaders still use the vanilla lightmaps, while others re-calculate lighting themselves. It's highly likely that your shader is using a hybrid of vanilla and custom lighting based off the held lights working. Try looking for any settings related to lighting that might do that, or try a few others like Complimentary Reimagined.

**Comment by AnOpenSauceDev** [comment]
it seems like quite a few shaders do have some confusion with methane, but it seems like the raytracing shaders I tested with originally are fine. This may have been an oversight on my end.

**Comment by AnOpenSauceDev** [comment]
100% a shader issue. I'm also getting some really weird lighting when using non-reimagined complementary

**Comment by silverace71** [comment]
i used reimagined in my testing, it was still bugged out

**Comment by silverace71** [comment]
1, you should make a discord so that its easier to troubleshoot (I can help you with it) and 2, lemme try stripping down my mods to only those essential and see if something happens with it

**Comment by AnOpenSauceDev** [comment]
im seeing the exact same issue as you with pretty barebones instance

**Comment by silverace71** [comment]
I mean i got raytracing and I can even see god rays, but on my version its still just dark

**Comment by silverace71** [comment]
https://pastebin.com/k3pFPVHf

heres an entire log if you can use it

**Comment by AnOpenSauceDev** [comment]
is it private? I'm getting a 404,

**Comment by silverace71** [comment]
odd I named it fortnite battle pass as a joke and i guess it got taken down?

**Comment by AnOpenSauceDev** [comment]
try putting it with a more generic name.

**Comment by silverace71** [comment]
https://pastebin.com/pg1CeXST

**Comment by silverace71** [comment]
that one should work hopefully

**Comment by AnOpenSauceDev** [comment]
Still 404. Try attaching it into github instead

**Comment by silverace71** [comment]
photon (shaders) seems to work slightly better but still has issue of enclosed spaces being dark

**Comment by silverace71** [comment]
aight one sec

**Comment by silverace71** [comment]
too big for comments, ummmmmmmmm. lemme make a repo lol

**Comment by silverace71** [comment]
https://raw.githubusercontent.com/silverace71/bugs/main/log-file.txt

**Comment by silverace71** [comment]
there we go. Now THAT should work

**Comment by silverace71** [comment]
oh, multimc has a upload function https://paste.ee/p/tT98y

i am dum

**Comment by AnOpenSauceDev** [comment]
seeing no actual errors related to methane, but i think this is due to how most shaders handle methane's weird lighting values

**Comment by silverace71** [comment]
![image](https://user-images.githubusercontent.com/77552030/212503953-77e5f7fb-69d4-4f4c-9a80-0004dc53c9a9.png)

This is why I like this mod

**Comment by silverace71** [comment]
also I could try and see if IMS might be able to find something on this (iris shaders discord)

**Comment by AnOpenSauceDev** [comment]
i think so, but this would probably result in iris detecting methane, and if so, changing the default lighting values.

**Comment by silverace71** [comment]
hey wanna make a discord server so that its easier to track things, and bug fix?

**Comment by AnOpenSauceDev** [comment]
i'm not really a discord user, and don't have enough time right now.

**Comment by silverace71** [comment]
fair, but also dude, look at this fps im getting
![image](https://user-images.githubusercontent.com/77552030/212504071-080b20cd-e758-48fc-9263-3e4cd2931135.png)

**Comment by AnOpenSauceDev** [comment]
This likely is derived from your CPU not needing to do all those lighting calculations

**Comment by silverace71** [comment]
i7 11700kf overclocked 128GB of ddr4 3600 mem and a 3070 and Im getting 3500 fps. Bruh

**Comment by silverace71** [comment]
wait a minute. Im testing this in 1.19.2. What if i go to 1.19.3. One sec lemme do that

**Comment by silverace71** [comment]
ay coderbot said
![image](https://user-images.githubusercontent.com/77552030/212504611-6f0f26c8-2419-4790-bc10-f825c88fc85a.png)

**Comment by AnOpenSauceDev** [comment]
makes sense

**Comment by AnOpenSauceDev** [comment]
the issue is that methane _is_ sending a skylight of 15, so this complicates things even more

**Comment by silverace71** [comment]
try makeing it so that it light updates only when you load into a world/server or place a block (not when a piston moves or water, stuff like that. Only when players break or place a block)

also you could make it so that it uses starlight when it does those updates

**Comment by AnOpenSauceDev** [comment]
a lot of that stuff is treated as block place/breaks, so this could be a problem

**Comment by AnOpenSauceDev** [comment]
after messing with this for a bit, i have found that the only reliable way would have to be removing most of methane's optimizations or doing #7 when shaders are on

**Comment by AnOpenSauceDev** [comment]
to be fixed with #7.

**Comment by AnOpenSauceDev** [comment]
@silverace71 fully fixed as of [1.5](https://modrinth.com/mod/methane/version/1.5)
